samples/Mvc.Server/Controllers/AuthorizationController.cs

```diff
@@ -0,0 +1,183 @@
+﻿/*
+ * Licensed under the Apache License, Version 2.0 (http://www.apache.org/licenses/LICENSE-2.0)
+ * See https://github.com/openiddict/openiddict-core for more information concerning
+ * the license and the contributors participating to this project.
+ */
+
+using System;
+using System.Security.Claims;
+using System.Threading;
+using System.Threading.Tasks;
+using AspNet.Security.OpenIdConnect.Extensions;
+using AspNet.Security.OpenIdConnect.Server;
+using Microsoft.AspNetCore.Authentication;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Http.Authentication;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.AspNetCore.Mvc;
+using Mvc.Server.Models;
+using Mvc.Server.ViewModels.Authorization;
+using OpenIddict;
+
+namespace Mvc.Server {
+    public class AuthorizationController : Controller {
+        private readonly OpenIddictApplicationManager<OpenIddictApplication<Guid>> _applicationManager;
+        private readonly SignInManager<ApplicationUser> _signInManager;
+        private readonly OpenIddictUserManager<ApplicationUser> _userManager;
+
+        public AuthorizationController(
+            OpenIddictApplicationManager<OpenIddictApplication<Guid>> applicationManager,
+            SignInManager<ApplicationUser> signInManager,
+            OpenIddictUserManager<ApplicationUser> userManager) {
+            _applicationManager = applicationManager;
+            _signInManager = signInManager;
+            _userManager = userManager;
+        }
+
+        [Authorize, HttpGet, HttpPost, Route("~/connect/authorize")]
+        public async Task<IActionResult> Authorize() {
+            // Note: when a fatal error occurs during the request processing, an OIDC response
```

Comment:
Yeah, that's the approach I tried in the latest commits.

I'll also give the status code middleware a try before merging this PR. It might give us more flexibility.